### ISSUE TITLE
pdf(d::Logistic, x::Real) is not safe!

### ISSUE THREAD

**Issue by janrpeters** [JuliaStats/Distributions.jl#781]
I am new to Julia but this bug affects porting my algorithms to Julia:
```julia
pdf(d::Logistic, x::Real) = (e = exp(-zval(d, x)); e / (d.θ * (1 + e)^2))
```
is not safe. For example, 
```julia
pdf(Logistic(6,0.01),-2)
```
will yield `NaN` instead of zero. The proper way to define this pdf is 
```julia
pdf(d::Logistic, x::Real) = logistic(-zval(d, x))*(1-logistic(-zval(d, x)))/d.θ;
```
which yields 0. 

May I directly patch this? Or can one of you do this? Thanks!
-Jan

**Comment by andreasnoack** [comment]
Please open a PR and include a test as well. It should probably be something like
```julia
z = zval(d, x)
lz = logistic(-abs(z))
return lz*(1 - lz)/d.θ
```
to get the highest accuracy and avoid computing special functions twice.

**Comment by Nosferican** [comment]
This should be closed due to #782.

**Comment by andreasnoack** [comment]
Fixed by #782
